Mathlib/GroupTheory/GroupAction/Hom.lean

```diff
@@ -13,201 +17,419 @@ import Mathlib.Algebra.Module.Basic
 
 ## Main definitions
 
-* `MulActionHom M X Y`, the type of equivariant functions from `X` to `Y`, where `M` is a monoid
-  that acts on the types `X` and `Y`.
-* `DistribMulActionHom M A B`, the type of equivariant additive monoid homomorphisms
-  from `A` to `B`, where `M` is a monoid that acts on the additive monoids `A` and `B`.
-* `MulSemiringActionHom M R S`, the type of equivariant ring homomorphisms
-  from `R` to `S`, where `M` is a monoid that acts on the rings `R` and `S`.
+* `MulActionHom φ X Y`, the type of equivariant functions from `X` to `Y`,
+  where `φ : M → N` is a map, `M` acting on the type `X` and `N` acting on the type of `Y`.
+* `DistribMulActionHom φ A B`,
+  the type of equivariant additive monoid homomorphisms from `A` to `B`,
+  where `φ : M → N` is a morphism of monoids,
+  `M` acting on the additive monoid `A` and `N` acting on the additive monoid of `B`
+* `SMulSemiringHom φ R S`, the type of equivariant ring homomorphisms
+  from `R` to `S`, where `φ : M → N` is a morphism of monoids,
+  `M` acting on the ring `R` and `N` acting on the ring `S`.
 
 The above types have corresponding classes:
-* `SMulHomClass F M X Y` states that `F` is a type of bundled `X → Y` homs
-  preserving scalar multiplication by `M`
-* `DistribMulActionHomClass F M A B` states that `F` is a type of bundled `A → B` homs
-  preserving the additive monoid structure and scalar multiplication by `M`
-* `MulSemiringActionHomClass F M R S` states that `F` is a type of bundled `R → S` homs
-  preserving the ring structure and scalar multiplication by `M`
+* `MulActionHomClass F φ X Y` states that `F` is a type of bundled `X → Y` homs
+  which are `φ`-equivariant
+* `DistribMulActionHomClass F φ A B` states that `F` is a type of bundled `A → B` homs
+  preserving the additive monoid structure and `φ`-equivariant
+* `SMulSemiringHomClass F φ R S` states that `F` is a type of bundled `R → S` homs
+  preserving the ring structure and `φ`-equivariant
 
 ## Notations
 
-* `X →[M] Y` is `MulActionHom M X Y`.
-* `A →+[M] B` is `DistribMulActionHom M A B`.
-* `R →+*[M] S` is `MulSemiringActionHom M R S`.
+* `X →ₑ[φ] Y` is `MulActionHom φ X Y`.
+* `A →ₑ+[φ] B` is `DistribMulActionHom φ A B`.
+* `R →ₑ+*[φ] S` is `MulSemiringActionHom φ R S`.
```

Comment:
Let's specify that the `ₑ` means *equivariant* somewhere.

Reply:
Done